modules/core/src/test/scala/org/tesselation/trust/DataGenerationSuite.scala

```diff
@@ -0,0 +1,20 @@
+package org.tesselation.trust
+
+import cats.effect.{Async, IO}
+
+import weaver.SimpleIOSuite
+import weaver.scalacheck.Checkers
+
+object DataGenerationSuite extends SimpleIOSuite with Checkers {
+  test("Data generation has values populated") {
+    val numNodes = 30
+    for {
+      _ <- Async[IO].pure(IO.pure(()))
```

Comment:
Why is that needed? If you have only one operation, you don't need to write a for comp and do such tricks at the beginning. Please consider following:

```scala
val numNodes = 30
val gen = new DataGenerator()

gen.generateData(numNodes)
  .map(_.head.edges)
  .flatMap { edges =>
    expect.all(edges.exists(_.trust != 0), edges.nonEmpty, edges.length < numNodes)
  }
```

Reply:
So just to clarify -- the Async came from the Concurrent from previous code base. All I did was move this over and get it working with new cats version. It required an implicit here. I actually don't think it matters if any of this code has async or anything else, since it's purely for generating test data, but it's what was there before and I need to base changes off it.

The code you posted in snippet above won't work -- because it's missing the implicit required by Ref for the random -- I tried moving the code over as is and it's missing an implicit for Sync provided by Async. (this is due to changing cats version.)

Reply:
<img width="1107" alt="image" src="https://user-images.githubusercontent.com/4959931/141795333-f7e40c6b-f22c-464e-a275-75ebf8daf2db.png">

<img width="747" alt="image" src="https://user-images.githubusercontent.com/4959931/141795377-c4673842-821e-4f6f-b0de-8d0a7f1c0f0f.png">

Reply:
@marcinwadon fixed